gene2phenotype_project/gene2phenotype_app/serializers/user.py

```diff
@@ -130,31 +180,252 @@ def create(self, validated_data):
                 - is_superuser: set to True if the user is a super user (default: False)
                 - is_staff: set to True if the user is a staff member (default: False)
         """
-        return User.objects.create_user(**validated_data)
 
+        user = User.objects.create_user(**validated_data)
+        if user:
+            CustomMail.send_create_email(data=user, subject="Account Created!", to_email=user.email)
+            return user 
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'first_name', 'last_name', 'password', 'password2', 'is_superuser', 'is_staff']
+        extra_kwargs = {'password': {'write_only': True, 'min_length': 6, 'max_length': 20}, 
+                        'email': {
+                            'validators': [
+                                UniqueValidator(
+                                    queryset=User.objects.all(),
+                                    message="This email is already registered"
+                                )
+                            ]
+                        }
+                        }
+        
+class ChangePasswordSerializer(serializers.ModelSerializer): 
+    """
+        Serializer class for Change Password
+
+        Args:
+            serializers (_type_): 
+                Fields:
+                    old_password : current password 
+                    password : new password
+                    password2 : confirm new password
+
+        Raises:
+            serializers.ValidationError: Raises if old password is not correct
+            serializers.ValidationError: Raises if new password and confirm new password do not match 
+            serializers.ValidationError: Raises if new and old password are the same 
+
+        Returns:
+            _type_: user email
+    """ 
+
+    old_password = serializers.CharField(max_length=20, min_length=6, style={'input_type' : 'password'}, write_only=True)
+    password = serializers.CharField(max_length=20, min_length=6, style={'input_type': 'password'}, write_only=True)
+    password2 = serializers.CharField(max_length=20, min_length=6, style={'input_type': 'password'}, write_only=True)
+
+    def validate(self, attrs):
+        """
+            Validation method for Change password method
+
+            Args:
+                attrs (_type_): Dictionary
+
+            Raises:
+                serializers.ValidationError: Raises if old password is not correct
+                serializers.ValidationError: Raises if new password and confirm new password do not match 
+
+            Returns:
+                _type_: Validated dictionary attrs 
+        """        
+        user = self.context.get('user')
+        old_password = attrs.get('old_password')
+        if user.check_password(old_password) is False:
+            raise serializers.ValidationError({'message' : "The password you entered is incorrect. Please provide the correct current password to update your password"})
```

Comment:
If I try to change someone else's password I get this message:
`"The password you entered is incorrect. Please provide the correct current password to update your password"`
It happens because this validation assumes the logged in user is the same as the username for which the password is trying to be changed.

Reply:
Yeah I do not send username/email as one of the serializers data to be sent when changing password